TACT.Net/SystemFiles/Encoding/EncodingFile.cs

```diff
@@ -289,14 +291,14 @@ public bool Remove(CASRecord record, TACTRepo tactRepo = null)
             if (_CKeyEntries.TryGetValue(record.CKey, out var entry))
             {
                 _CKeyEntries.Remove(record.CKey);
-                _EKeyEntries.Remove(entry.EKey);
+                entry.EKeys.ForEach(x => _EKeyEntries.Remove(x));
 
                 // propagate removal
                 if (tactRepo != null)
                 {
-                    tactRepo.IndexContainer?.Remove(entry.EKey);
-                    tactRepo.DownloadFile?.Remove(entry.EKey);
-                    tactRepo.DownloadSizeFile?.Remove(entry.EKey);
+                    entry.EKeys.ForEach(x => tactRepo.IndexContainer?.Remove(x));
+                    entry.EKeys.ForEach(x => tactRepo.DownloadFile?.Remove(x));
+                    entry.EKeys.ForEach(x => tactRepo.DownloadSizeFile?.Remove(x));
```

Comment:
A bit pedantic but can you change this to iterate once please, it just saves on the loops?  e.g.
```c#
entry.EKeys.ForEach(x =>
{
    _EKeyEntries.Remove(x);

   // propagate removal
   if (tactRepo != null) {
      ...
   }
});
```
Do you know if EKeys are now shared across CKeyEntries? If so we may need to change this.

Reply:
> Do you know if EKeys are now shared across CKeyEntries? If so we may need to change this.

I have no idea, but I will change the suggested code.